cmd/list.go

```diff
@@ -80,12 +82,25 @@ func doList(cmd *cobra.Command, args []string) error {
 				} else {
 					v = strconv.Quote(string(value))
 				}
-
-				fmt.Fprintln(w, strings.Join([]string{secret.Name, string(secret.Type), key, v}, "\t"))
+				formattedSecrets = append(formattedSecrets, []string{secret.Name, string(secret.Type), key, v})
 			}
 		}
 	}
 
+	// sort by KEY
+	sort.Slice(formattedSecrets, func(i, j int) bool {
+		return formattedSecrets[i][2] < formattedSecrets[j][2]
+	})
+
+	// sort by NAME
+	sort.SliceStable(formattedSecrets, func(i, j int) bool {
+		return formattedSecrets[i][0] < formattedSecrets[j][0]
+	})
+
+	for _, secret := range formattedSecrets {
+		fmt.Fprintln(w, strings.Join(secret, "\t"))
+	}
```

Comment:
Response of `ListSecrets` is already sorted alphabetically (see https://github.com/kubernetes/kubernetes/blob/v1.6.6/staging/src/k8s.io/apiserver/pkg/storage/etcd/etcd_helper.go#L456-L460 how k8s retrieves objects).
So we don't have to sort by key again!

Why don't you sort and print key-values in `for _, secret := range secrets.Items { ... }` loop?